controllers/operator/nmstate_controller.go

```diff
@@ -237,6 +239,34 @@ func (r *NMStateReconciler) applyHandler(instance *nmstatev1.NMState) error {
 	return r.renderAndApply(instance, data, "handler", true)
 }
 
+// webhookReplicaCount returns the number of replicas for the nmstate webhook
+// deployment based on the underlying infrastructure toplogy. It returns 2
+// values (and error):
+// 1. min. number of replicas
+// 2. number of desired replicas
+// 3. error
+func (r *NMStateReconciler) webhookReplicaCount(nodeSelector map[string]string) (int, int, error) { //nolint:gocritic
+	const (
+		multiNodeClusterReplicaCountDesired = 2
+		multiNodeClusterReplicaMinCount     = 1
+
+		singleNodeClusterReplicaCountDesired = 1
+		singleNodeClusterReplicaMinCount     = 0
+	)
+
+	nodes := corev1.NodeList{}
+	err := r.Client.List(context.TODO(), &nodes, client.MatchingLabels(nodeSelector))
+	if err != nil {
+		return 0, 0, fmt.Errorf("failed to get nodes: %w", err)
+	}
```

Comment:
The webhook is deployed only on the control-plane so this has to count number of control plane nodes excluding workers.

Reply:
So does the `nodeSelector` label match to worker nodes too?
https://github.com/nmstate/kubernetes-nmstate/blob/847d80edc8ee3828c52caad39d4fd05f88200b7b/controllers/operator/nmstate_controller.go#L194-L201